LucidGloves/src/DeviceProvider.cpp

```diff
@@ -29,8 +29,20 @@ VRDeviceConfiguration_t DeviceProvider::GetConfiguration(vr::ETrackedControllerR
 	const float offsetY = vr::VRSettings()->GetFloat(c_settingsSection, "y_offset");
 	const float offsetZ = vr::VRSettings()->GetFloat(c_settingsSection, "z_offset");
 
-	//x axis is flipped for the different hands
-	const vr::HmdVector3_t offsetVector = { role == vr::TrackedControllerRole_RightHand ? offsetX : -offsetX, offsetY, offsetZ };
+	const float offsetXDeg = vr::VRSettings()->GetFloat(c_settingsSection, "x_offsetDegrees");
+	const float offsetYDeg = vr::VRSettings()->GetFloat(c_settingsSection, "y_offsetDegrees");
+	const float offsetZDeg = vr::VRSettings()->GetFloat(c_settingsSection, "z_offsetDegrees");
+
+	const bool leftReversedX = vr::VRSettings()->GetBool(c_settingsSection, "leftFlippedX");
+	const bool leftReversedY = vr::VRSettings()->GetBool(c_settingsSection, "leftFlippedX");
+	const bool leftReversedZ = vr::VRSettings()->GetBool(c_settingsSection, "leftFlippedX");
+
+	//x axis may be flipped for the different hands
+	const vr::HmdVector3_t offsetVector = { role == vr::TrackedControllerRole_RightHand || !leftReversedX ? offsetX : -offsetX, offsetY, offsetZ };
+	const vr::HmdVector3_t angleOffsetVector =
+	  { role == vr::TrackedControllerRole_RightHand || !leftReversedX ? offsetXDeg : -offsetXDeg, 
+		role == vr::TrackedControllerRole_RightHand || !leftReversedY ? offsetYDeg : -offsetYDeg,
+		role == vr::TrackedControllerRole_RightHand || !leftReversedZ ? offsetZDeg : -offsetZDeg };
```

Comment:
nitpick, but having a single `bool` for right handed-ness might make the logic easier to follow, as well as some brackets.